includes/Checker/Default_Check_Repository.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Default_Check_Repository
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use Exception;
+
+/**
+ * Default Check Repository class.
+ *
+ * @since n.e.x.t
+ */
+class Default_Check_Repository implements Check_Repository {
+
+	/**
+	 * Array map holding all runtime checks.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	protected $runtime_checks = array();
+
+	/**
+	 * Array map holding all runtime checks.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	protected $static_checks = array();
+
+	/**
+	 * Registers a check to the repository.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug  The checks slug.
+	 * @param Check  $check The Check instance.
+	 *
+	 * @throws Exception Thrown if Check does not use correct interface, or slug already exists.
+	 */
+	public function register_check( $slug, Check $check ) {
+		if ( ! $check instanceof Runtime_Check && ! $check instanceof Static_Check ) {
+			throw new Exception( __( 'Check must be an instance of Runtime_Check or Static_Check.', 'plugin-check' ) );
+		}
+
+		if ( array_key_exists( $slug, $this->runtime_checks ) || array_key_exists( $slug, $this->static_checks ) ) {
```

Comment:
Nit-pick: `isset()` is faster than `array_key_exists()`, and since the entries would never be `null`, there's no functional value in using `array_key_exists()` either.